project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/ontology/TIBTerminologyServiceIntegration.java

```diff
@@ -174,9 +180,12 @@ public List<OntologyClass> query(String searchTerm, int offset, int limit)
   @Override
   public Optional<OntologyClass> searchByCurie(String curie) throws LookupException {
     try {
-      return searchByOboIdExact(curie).map(TIBTerminologyServiceIntegration::convert);
+      return searchByOboIdExact(curie).map(this::updateCache)
+          .map(TIBTerminologyServiceIntegration::convert);
     } catch (IOException e) {
-      throw wrapIO(e);
+      // this happens on network interrupts or if the remote service is down
+      // we try to recover from the cache
+      return cache.findByCurie(curie).map(TIBTerminologyServiceIntegration::convert);
```

Comment:
You handle the exception here. It would be good to know that it occurred otherwise we are not informed of it. Maybe add a log entry here (warn or error)